File: 0035-search-insert-position/0035-search-insert-position.py
class Solution(object):
    def searchInsert(self, nums, target):
        l = 0
        h = len(nums) - 1
        while (l <= h):
            m = h + l/2
            if (nums[m] == target):
                return m
            elif (nums[m] > target):
                h = m - 1
            else:
                l = m + 1
        return (h+1)
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        

File: 0035-search-insert-position/0035-search-insert-position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        int r = nums.length-1;
        if (target < nums[0]){
            return 0;
        }
        if (target > nums[r]){
            return r+1;
        }
        int l = 0;
        while (l <= r){
            int m = (l+r) / 2;
            if (nums[m] == target){
                return m;
            }
            else if (nums[m] < target){
                l = m+1;
            }
            else{
                r = m - 1;
            }
        }
        return r+1;
    }
}